rmgpy/molecule/molecule.py

```diff
@@ -1595,4 +1595,38 @@ def toGroup(self):
         
         return group
 
-
+    def getNthNeighbor(self, startingAtoms, n, blacklist):
+        '''
+        Recursively get the Nth nonHydrogen neighbors of the startingAtoms.
+        StartingAtoms should be a list.
+        Return a list.
+        '''
+        neighbors = []
+        for atom in startingAtoms:
+            for neighbor in self.getBonds(atom):
```

Comment:
Nice tests. The list comprehension version and the current version are much cleaner for me than the two for loops version. Based on the tests, which show the current version is slower, I tend to choose the list comprehension version.